x-pack/plugins/observability_ai_assistant/server/functions/esql/correct_common_esql_mistakes.test.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import dedent from 'dedent';
+import { noop } from 'lodash';
+import { correctCommonEsqlMistakes } from './correct_common_esql_mistakes';
+
+describe('correctCommonEsqlMistakes', () => {
+  const fakeLogger = {
+    debug: noop,
+  } as any;
+
+  function renderQuery(query: string) {
+    return '```esql\n' + dedent(query) + '\n```';
+  }
+
+  function expectQuery(input: string, expectedOutput: string) {
+    expect(correctCommonEsqlMistakes(renderQuery(input), fakeLogger)).toEqual(
+      renderQuery(expectedOutput)
+    );
+  }
+
+  it('replaces aliasing via the AS keyword with the = operator', () => {
+    expectQuery(`FROM logs-* | STATS COUNT() AS count`, 'FROM logs-*\n| STATS count = COUNT()');
+    expectQuery(
+      `FROM logs-* | STATS AVG(transaction.duration.histogram) AS avg_request_latency, PERCENTILE(transaction.duration.histogram, 95) AS p95`,
+      `FROM logs-*
+      | STATS avg_request_latency = AVG(transaction.duration.histogram), p95 = PERCENTILE(transaction.duration.histogram, 95)`
+    );
+
+    expectQuery(
+      `FROM traces-apm*
+      | WHERE @timestamp >= NOW() - 24 hours
+      | STATS AVG(transaction.duration.us) AS avg_duration, SUM(success) / (SUM(success) + SUM(failure)) AS success_rate, COUNT(*) AS total_requests BY service.name`,
+      `FROM traces-apm*
```

Comment:
I think this query is not correct, it is one of the errors I saw in the tests. STATS does not accept complex operations:
- This query: `from .kibana-observability-ai-assistant-kb-000001 | STATS count = count(id)/count_distinct(doc_id) by doc_id` -> throws this error `expected an aggregate function or group but got [count(id)/count_distinct(doc_id)] of type [Div]`
- The query that works is: `from .kibana-observability-ai-assistant-kb-000001 | STATS count = count(id), distinct = count_distinct(doc_id) by doc_id | eval count/distinct`

Reply:
cheers, it doesn't really matter here but I've corrected the query